### ISSUE TITLE
Finetuning returning strange results ( <1> <0> <0> <0> <0>...)

### ISSUE THREAD

**Issue by patrusso2** [google-research/pegasus#79]
I used my own finetuning dataset and my validation samples look great after 1k or 2k training steps but it soon devolves to predicting:

 <1> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0> <0>  

I'm using Multi News as the starting checkpoint, but since I can only use batch size 1 I lowered the training rate by a factor of 10 (to reduce noise).

My text encoding is unicode, but the text is in English with some special characters (e.g., curly quotes).  My understanding is that I should leave in uncommon words/characters and should leave the text cased.  Did I misunderstand and should I be doing that preprocessing myself?  If it's not a preprocessing issue would you have an idea of what else could be off?

Thanks!

P.S. I know it could be data/model specific so if no one has immediate insight I'll continue experimenting and report back - but it seems like it could be something others have seen.  Can close this if it's not appropriate here!

**Comment by JingqingZ** [comment]
Hi, this may be caused by many potential reasons so I am not sure if I can give you a solution. It may worth checking the output from tokenization (sentencepiece) if it is reasonable. Or if the learning rate is too high or low. Sometimes, the beam size or beam alpha can make a difference.

Unicode or uncommon words/chars should be fine. We didn't do any special preprocessing of data besides tokenization.

**Comment by patrusso2** [comment]
Thanks @JingqingZ  I really appreciate the feedback and help that you and your collaborators are providing.  The assistance here and on other issues has been super helpful!

Since unicode and uncommon chars should be fine I looked deeper through many individual examples and found that 1% of my training sample had empty targets.  It wasn't enough for me to notice a huge chunk missing, but (I'd guess especially with a batch size 1 ) it's enough to throw the model.  Completely my mistake!  

Thank you again!
